### PR TITLE
RESOURCE-418 AWS::ApiGatewayV2::Api

### PR DESCRIPTION
**Pull request by soumyo13**
Signed-off-by: Soumyodeep Karmakar <soumyo.k13@gmail.com>

### Description

1. It contains all the singular and plural resources.
2. Added the resource_id in the singular resource.
3. Modified the docs according to the new format. 

### Issues Resolved
This is a new resource. There are no issues to be fixed.

### Check List
Please fill a box or appropriate ([x]) or mark N/A.
- [x] New functionality includes integration tests/controls
- [x] New Terraform resources
- [x] Documentation provided or updated for resources 
- [x] All Integration Tests pass
- [x] All Unit Tests pass
- [x] `rake lint` passes
- [x] All commits have been signed off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

**Review by sa-progress — APPROVED** [review]

File: docs-chef-io/content/inspec/resources/aws_api_gateway_v2_api.md [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+
+`tags`
+: A collection of tags associated with the API.
+
+`version`
+: A version identifier for the API.
+
+`warnings`
+: The warning messages reported when `failonwarnings` is turned on during API import.
+
```
Comment (dkumaras) [inline docs-chef-io/content/inspec/resources/aws_api_gateway_v2_api.md:103]: Also, one global query - dont we have to mention about the property type - string, integer, boolean, etc ?

**Review by dkumaras — COMMENTED** [review]
Pls chk n confirm. Also left a couple of queries/suggestions.

File: docs-chef-io/content/inspec/resources/aws_api_gateway_v2_api.md [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+: Specifies whether clients can invoke your API by using the default execute-api endpoint. By default, clients can invoke your API with the default https://{api_id}.execute-api.{region}.amazonaws.com endpoint. To require that clients use a custom domain name to invoke your API, disable the default endpoint.
+
+`import_info`
+: The validation information during API import.
+
+`name`
+: The name of the API.
+
+`protocol_type`
+: The API protocol.
```
[the remark this answers is not in the document]
Comment (soumyo13) [reply docs-chef-io/content/inspec/resources/aws_api_gateway_v2_api.md]: Yeah

File: docs-chef-io/content/inspec/resources/aws_api_gateway_v2_api.md (hunk 1 above)
Comment (soumyo13) [reply docs-chef-io/content/inspec/resources/aws_api_gateway_v2_api.md:103]: We can add but this change must be done to all the resources which will be a lot of work.

Comment (dkumaras) [reply docs-chef-io/content/inspec/resources/aws_api_gateway_v2_api.md:103]: Ok, lets once check with Clinto too!

File: docs-chef-io/content/inspec/resources/aws_api_gateway_v2_api.md [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+: The route selection expression for the API. For HTTP APIs, the routeSelectionExpression must be ${request.method} ${request.path}. If not provided, this will be the default for HTTP APIs. This property is required for WebSocket APIs.
+
+`tags`
+: A collection of tags associated with the API.
+
+`version`
+: A version identifier for the API.
+
+`warnings`
+: The warning messages reported when `failonwarnings` is turned on during API import.
```
[the remark this answers is not in the document]
Comment (soumyo13) [reply docs-chef-io/content/inspec/resources/aws_api_gateway_v2_api.md]: 'failonwarnings' specifies whether to rollback the API creation when a warning is encountered. By default, API creation continues if a warning is encountered.

We can add this.

**Review by dkumaras — COMMENTED** [review]
Pls chk n confirm to approve it from doc side.

File: docs-chef-io/content/inspec/resources/aws_api_gateway_v2_api.md [hunk 4]
```diff
@@ -98,6 +98,11 @@ end
 `version`
 : A version identifier for the API.
 
+`failonwarnings`
+: Specifies to rollback the API creation when a warning is encountered. By default, API creation continues if a warning is encountered.
+
+: **Field**: `failonwarnings`
```
Comment (soumyo13) [inline docs-chef-io/content/inspec/resources/aws_api_gateway_v2_api.md:104]: Please remove this. There is no attribute named failonwarnings. 

https://docs.aws.amazon.com/sdk-for-ruby/v3/api/Aws/ApiGatewayV2/Client.html#get_api-instance_method

File: docs-chef-io/content/inspec/resources/aws_api_gateway_v2_apis.md [hunk 5]
```diff
@@ -109,6 +109,11 @@ This resource does not require any parameters.
 
 : **Field**: `version`
 
+`failonwarnings`
+: Specifies to rollback the API creation when a warning is encountered. By default, API creation continues if a warning is encountered.
+
+: **Field**: `failonwarnings`
```
Comment (soumyo13) [inline docs-chef-io/content/inspec/resources/aws_api_gateway_v2_apis.md:115]: Remove from here also.

**Review by dkumaras — APPROVED** [review]

**Pull request merged**